azkaban-common/src/main/java/azkaban/DispatchMethod.java

```diff
@@ -0,0 +1,14 @@
+package azkaban;
+
+import azkaban.utils.Props;
+
+public enum DispatchMethod {
+    PUSH,
+    POLL,
+    CONTAINERIZED;
+
+    public static DispatchMethod getDispatchModel(Props azkabanProps) {
+        return DispatchMethod.valueOf(
+                azkabanProps.getString(Constants.ConfigurationKeys.AZKABAN_EXECUTION_DISPATCH_METHOD, "PUSH").toUpperCase());
```

Comment:
Should the default be Poll, considering that is what is used in Azkaban these days?

Reply:
I feel like the `getDispatchModel` method would be better suited for a service/"server initialization" class so that enum class doesn't need to know about the existence of an Azkaban properties file, it's just a data type for me. What do you think?

Reply:
@ypadron-in Let us have two methods. One is for service and another one for property. The reason is I have seen couple of places where property for poll is used in if cases.
@djaiswal83 I have kept PUSH as default because currently if poll is not set deliberately then PUSH is selected by default. So that will maintain consistency with existing behavior.